### ISSUE TITLE
Pixiv Support

### ISSUE THREAD

**Issue by SaltSouls** [laylavish/uBlockOrigin-HUGE-AI-Blocklist#10]
Not sure how hard this would be, but Pixiv has an AI tag that this may be able to take advantage of in order to block said images/creators from showing that post AI images.

**Comment by laylavish** [comment]
I'm not sure if this is possible, since i'm pretty sure uBO only has access to CSS selectors, not to the actual page data like what Google or DuckDuckGo has. Using something like `-intext:"AI-generated` in your search query (for Google Search) actually crawls for webpages without the Pixiv tag `AI-generated`. Unfortunately I think this isn't possible with plain CSS selectors, as there is not enough information to do that with since information like tags and descriptions are usually truncated.

**Comment by SaltSouls** [comment]
Yeah I mean on the pixiv site itself, it has a full tag for AI-Generated images and the like.
I don't know if uBlock is able to take advantage of it, but I figure if it could, then it would be nice to filter out all the slop when searching for images there.

**Comment by laylavish** [comment]
So I've looked into this a bit, and the closest thing to the tags pixiv has is the alt text, which can house tags. Problem is, is that alt text is user-specified; it isn't actually tied to the tags system at all. The tags do not appear in html (when searching for images, eg. scrolling homepage, search bar queries, etc), so uBlock can't take advantage of them. Pretty unfortunate.

I will say however that Pixiv does have an AI-generated content toggle in `Display settings` in your Pixiv settings, which would hide any AI content that is tagged with `AI-generated`. I could be wrong about this, but I believe the AI-generated content toggle is off by default, meaning that all AI slop should be hidden.

**Comment by MonoS** [comment]
> I did something similar...

I wrote a possible solution using another plugin, firefox only, it was hidden up until now as my account was being limited.
